### PR TITLE
Add error handling to the buybutton

### PR DESCRIPTION
**Pull request by salesfelipe**
#### What is the purpose of this pull request?
Increase the robustness of the buybutton component

#### What problem is this solving?
Before when it couldn't retrieve the orderFormId it would break without any feedback to the user.

#### How should this be manually tested?
[workspace](https://felipe--storecomponents.myvtex.com/)

#### Screenshots or example usage
image

#### Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/buy-button/index.js [hunk 1]
```diff
@@ -15,22 +18,43 @@ export class BuyButton extends Component {
...
     seller: 1,
   }
+
+  translateMessage = id => this.props.intl.formatMessage({ id: id })
+
+  toastMessage = success => {
+    const event = new Event(EVENT_TOAST)
+
+    if (success) {
+      event.detail = {
```
Comment (brunojdo) [inline react/components/buy-button/index.js]: How about Change this for a ternary operator?

File: react/components/buy-button/index.js [hunk 2]
```diff
@@ -42,36 +66,38 @@ export class BuyButton extends Component {
...
-    }, (err) => {
-      if (err) {
-        document.dispatchEvent(new Event('item:fail'))
-        this.setState({ isLoading: !this.state.isLoading })
+    }).then(
+      res => {
+        const { items } = res.data.addItem
+
+        if (find(items, { id: skuId })) {
+          this.toastMessage(true)
```
Comment (brunojdo) [inline react/components/buy-button/index.js]: idem

File: react/components/buy-button/index.js [hunk 3]
```diff
@@ -42,36 +66,38 @@ export class BuyButton extends Component {
...
+        } else {
+          this.toastMessage(false)
+        }
+
+        this.setState({ isLoading: false })
+      },
+      () => {
+        this.toastMessage(false)
+        this.setState({ isLoading: false })
```
Comment (brunojdo) [inline react/components/buy-button/index.js]: Would be better also passtrough the `err` of the callback.

Comment (salesfelipe) [reply react/components/buy-button/index.js]: 👍

File: react/components/buy-button/index.js [hunk 4]
```diff
@@ -42,36 +62,34 @@ export class BuyButton extends Component {
...
-        this.setState({ isLoading: !this.state.isLoading })
-      }
-    }, (err) => {
-      if (err) {
-        document.dispatchEvent(new Event('item:fail'))
-        this.setState({ isLoading: !this.state.isLoading })
+    }).then(
+      res => {
+        const { items } = res.data.addItem
+
```
Comment (brunojdo) [inline react/components/buy-button/index.js]: remove blank lines ⛏

File: react/components/BuyButton/CHANGELOG.md [hunk 5]
```diff
@@ -7,53 +7,59 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
```
Comment (brunojdo) [inline react/components/BuyButton/CHANGELOG.md]: Remove this CHANGELOG

Comment (salesfelipe) [reply react/components/BuyButton/CHANGELOG.md]: 👍

**Pull request merged**